### PR TITLE
Adjust the mobile-app-config to support the new data service

### PR DESCRIPTION
**Pull request by sorin-iovita**
## Description

This PR includes:

- Addition of new application properties in order to be passed to the app-config.
- Environment variables to replace application properties at runtime
- Necessary mappings from yaml to protobuf

### CODE REVIEW

File: services/submission/src/test/java/app/coronawarn/server/services/submission/validation/ValidSubmissionPayloadTest.java [hunk 1]
```diff
@@ -17,21 +17,23 @@
...
 import org.springframework.http.ResponseEntity;
 import org.springframework.test.annotation.DirtiesContext;
-import org.springframework.test.context.ActiveProfiles;
 
 /**
  * This test must have the public modifier to be callable by the interpolation logic.
  */
 @SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
 @DirtiesContext
+@ExtendWith(MockitoExtension.class)
```
Comment (EugenM-SAP) [inline services/submission/src/test/java/app/coronawarn/server/services/submission/validation/ValidSubmissionPayloadTest.java:36]: Was this extend annotation needed now for mocking capabilities?

**Review by EugenM-SAP — DISMISSED** [review]

**Review by ioangut — DISMISSED** [review]

File: services/distribution/src/main/resources/application.yaml [hunk 2]
```diff
@@ -131,6 +131,32 @@ services:
...
+        hours-since-test-to-submit-key-submission-metadata: ${PPA_HOURS_SINCE_TEST_RESULT_TO_SUBMIT_KEY_SUBMISSION_METADATA:36}
+      android-privacy-preserving-analytics-parameters:
+        probability-to-submit: ${PPA_PROBABILITY_TO_SUBMT:1}
+        probability-to-submit-exposure-windows: ${PPA_PROBABILOITY_TO_SUBMIT_EXPOSURE_WINDOWS:1}
+        hours-since-test-registration-to-submit-test-result-metadata: ${PPA_HOURS_SINCE_TEST_RESULT_TO_SUBMIT_KEY_RESULT_METADATA:165}
+        hours-since-test-to-submit-key-submission-metadata: ${PPA_HOURS_SINCE_TEST_RESULT_TO_SUBMIT_KEY_SUBMISSION_METADATA:36}
+        require-basic-integrity: ${PPA_PPAC_REQUIRE_BASIC_INTEGRITY:false}
+        require-cts-profile-match: ${PPA_PPAC_REQUIRE_CTS_PROFILE_MATCH:false}
+        require-evaluation-type-basic: ${PPA_PPAC_REQUIRE_EVALUATION_TYPE_BASIC:false}
+        require-evaluation-type-hardware-backed: ${PPA_PPAC_REQUIRE_EVALUATION_TYPE_HARDWARE_BACKED:false}
```
Comment (hilmarf) [inline services/distribution/src/main/resources/application.yaml:159]: @mlenkeit - please have a look at the defaults! All good?

File: services/distribution/src/main/resources/application.yaml [hunk 3]
```diff
@@ -131,6 +131,32 @@ services:
...
         max-exposure-detections-per-interval: ${ANDROID_EXPOSURE_DETECTION_MAX_ED_PER_INTERVAL:6}
         overall-timeout-in-seconds: ${ANDROID_EXPOSURE_DETECTION_OVERALL_TIMEOUT_IN_SECONDS:900}
+      ios-event-driven-user-survey-parameters:
+        otp-query-parameter-name: ${EDUS_OTP_QUERY_PARAMETER_NAME:otp}
+        survey-on-high-risk-enabled: ${EDUS_SURVEY_ON_HIGH_RISK_ENABLED:true}
+        survey-on-high-risk-url: ${EDUS_SURVEY_ON_HIGH_RISK_URL}
+      android-event-driven-user-survey-parameters:
+        otp-query-parameter-name: ${EDUS_OTP_QUERY_PARAMETER_NAME:otp}
+        survey-on-high-risk-enabled: ${EDUS_SURVEY_ON_HIGH_RISK_ENABLED:true}
+        survey-on-high-risk-url: ${EDUS_SURVEY_ON_HIGH_RISK_URL}
```
Comment (hilmarf) [inline services/distribution/src/main/resources/application.yaml:141]:
```suggestion
        survey-on-high-risk-url: ${EDUS_SURVEY_ON_HIGH_RISK_URL:https://befragungen.rki.de/CWABasisbefragung}
```

File: services/distribution/src/main/resources/application.yaml [hunk 4]
```diff
@@ -131,6 +131,32 @@ services:
       android-exposure-detection-parameters:
         max-exposure-detections-per-interval: ${ANDROID_EXPOSURE_DETECTION_MAX_ED_PER_INTERVAL:6}
         overall-timeout-in-seconds: ${ANDROID_EXPOSURE_DETECTION_OVERALL_TIMEOUT_IN_SECONDS:900}
+      ios-event-driven-user-survey-parameters:
+        otp-query-parameter-name: ${EDUS_OTP_QUERY_PARAMETER_NAME:otp}
+        survey-on-high-risk-enabled: ${EDUS_SURVEY_ON_HIGH_RISK_ENABLED:true}
+        survey-on-high-risk-url: ${EDUS_SURVEY_ON_HIGH_RISK_URL}
```
Comment (hilmarf) [inline services/distribution/src/main/resources/application.yaml:137]:
```suggestion
        survey-on-high-risk-url: ${EDUS_SURVEY_ON_HIGH_RISK_URL:https://befragungen.rki.de/CWABasisbefragung}
```

**Review by hilmarf — DISMISSED** [review]

File: services/submission/src/test/java/app/coronawarn/server/services/submission/validation/ValidSubmissionPayloadTest.java (hunk 1 above)
Comment (sorin-stefan-iovita-sap) [reply services/submission/src/test/java/app/coronawarn/server/services/submission/validation/ValidSubmissionPayloadTest.java:36]: SonarCloud was complaining about missing mock initialization as a bug and one of the proposed solution was this one.

**Review by hilmarf — DISMISSED** [review]

File: services/distribution/src/main/resources/application.yaml [hunk 5]
```diff
@@ -131,6 +131,32 @@ services:
       android-exposure-detection-parameters:
         max-exposure-detections-per-interval: ${ANDROID_EXPOSURE_DETECTION_MAX_ED_PER_INTERVAL:6}
         overall-timeout-in-seconds: ${ANDROID_EXPOSURE_DETECTION_OVERALL_TIMEOUT_IN_SECONDS:900}
+      ios-event-driven-user-survey-parameters:
+        otp-query-parameter-name: ${EDUS_OTP_QUERY_PARAMETER_NAME:otp}
+        survey-on-high-risk-enabled: ${EDUS_SURVEY_ON_HIGH_RISK_ENABLED:true}
+        survey-on-high-risk-url: ${EDUS_SURVEY_ON_HIGH_RISK_URL:https://befragungen.rki.de/CWABasisbefragung}
+      android-event-driven-user-survey-parameters:
+        otp-query-parameter-name: ${EDUS_OTP_QUERY_PARAMETER_NAME:otp}
+        survey-on-high-risk-enabled: ${EDUS_SURVEY_ON_HIGH_RISK_ENABLED:true}
```
Comment (mlenkeit) [inline services/distribution/src/main/resources/application.yaml:140]:
```suggestion
        survey-on-high-risk-enabled: ${EDUS_SURVEY_ON_HIGH_RISK_ENABLED:false}
```

Comment (mlenkeit) [reply services/distribution/src/main/resources/application.yaml:140]: Should be false by default (I'll update the spec) and true on DEV, INT, WRU

File: services/distribution/src/main/resources/application.yaml [hunk 6]
```diff
@@ -131,6 +131,32 @@ services:
       android-exposure-detection-parameters:
         max-exposure-detections-per-interval: ${ANDROID_EXPOSURE_DETECTION_MAX_ED_PER_INTERVAL:6}
         overall-timeout-in-seconds: ${ANDROID_EXPOSURE_DETECTION_OVERALL_TIMEOUT_IN_SECONDS:900}
+      ios-event-driven-user-survey-parameters:
+        otp-query-parameter-name: ${EDUS_OTP_QUERY_PARAMETER_NAME:otp}
+        survey-on-high-risk-enabled: ${EDUS_SURVEY_ON_HIGH_RISK_ENABLED:true}
```
Comment (mlenkeit) [inline services/distribution/src/main/resources/application.yaml:136]:
```suggestion
        survey-on-high-risk-enabled: ${EDUS_SURVEY_ON_HIGH_RISK_ENABLED:false}
```

Comment (mlenkeit) [reply services/distribution/src/main/resources/application.yaml:136]: Should be false by default (I'll update the spec) and true on DEV, INT, WRU

**Review by hilmarf — APPROVED** [review]

**Review by ioangut — APPROVED** [review]

**Pull request merged**